Descent3/terrainrender.cpp

```diff
@@ -2035,13 +2035,13 @@ void DrawSky(vector *veye, matrix *vorient) {
   for (i = 0; i < 6; i++) {
     for (t = 0; t < MAX_HORIZON_PIECES; t++) {
       tempvec = Terrain_sky.horizon_vectors[t][i];
-      tempvec.y -= veye->y * 0.5f;
+      tempvec.y() -= veye->y() * 0.5f;
       vm_MatrixMulVector(&Temp_sky_vectors[t][i], &tempvec, vorient);
 
       tempvec = Terrain_sky.horizon_vectors[t][i];
-      tempvec.x += veye->x;
-      tempvec.z += veye->z;
-      tempvec.y += veye->y * 0.5f;
+      tempvec.x() += veye->x();
```

Comment:
why is `x` a functor now? This is all getting very complicated, is there a way to keep the same API as before?

Reply:
The only way to achieve it is to copy std::array implementation, add
```cpp
union {
      typename __array_traits<_Tp, _Nm>::_Type _M_elems;
      struct { T x,y,z,w };
      struct { T h,p,b };
};
```
and use modified array as a base class

Reply:
probably with requires or optional members

Reply:
Another possibility might be CLANG only Vector Extensions but they don't allow scalar aligned 3 component types
or CUDA types which allow 3 component scalar/vector aligned types.

Reply:
ok, the user side of our vector API is fine as it stands, we can keep it this way